### DISCUSSION TITLE
Cppcheck, Intellisense and Multi Project Workspace

### DISCUSSION THREAD

**Discussion by AAsyunkin-se** [github0null/eide#230] [Q&A]
Hi all,

I have set up a multiproject as described in [https://em-ide.com/docs/advance/multi_prj_workspace](url)
I have Common code and 2 different target cpu code - 3 folders in total.
I also have Debug and Release config set up for each target cpu.
I can set Active Project as required (green dot in EIDE Projects pane) - all works as expected.

Now, there are Preprocessor Defines which are different for each target, and my Common code uses these Defines to compile correctly for each target.

The problem is with viewing the source files only - Intellisense with default settings (i.e. without setting up c_cpp_properties.json file because it is not recommended) is not aware of Preprocessor Defines and dims the code that's actually used for the Project set as Active.

Is there a way to have a specific Define set that will be passed to Intellisense based on which project is Active?

**Comment by github0null** [comment]
Sorry i'm late.

> Is there a way to have a specific Define set that will be passed to Intellisense based on which project is Active?

I'm not sure what you mean

The plugin will provide the configuration for each file, please check the log, please point out the incorrect place, and the behavior you expect

![image](https://github.com/github0null/eide/assets/43022536/679c9cab-84df-4a0f-9bb1-373f7d6a4e71)

**Comment by AAsyunkin-se** [comment]
![image](https://github.com/github0null/eide/assets/55350687/d4911d6a-3860-43fe-80f0-a0cb20d748f6)

CustomConfigurationProvider is run one time only upon VS Code startup. If I switch my Active project, then Include paths, Defines etc are not updated as per newly activated project. See attached screenshot - define (see red marks) there can't be resolved by pressing F12, and other defines that are project specific (e.g. Firmware in a different include path per project) - these are resolved as per 'first active project'.

**Comment by github0null** [comment]
This is a bug

I have published a preview version: `v3.12.2023051401`, you can update and try it again

**Reply by AAsyunkin-se** [reply]
Great news, thanks! I tried it and it works. Well...mostly :)
Defines and includes are resolved correctly as per active subproject.
Only one minor problem - if I open a file with different extension (i.e. not a 'c' or 'h') and then close it, VS Code will switch to the previous file window, and inactive code is no longer dimmed until i switch to another 'c' file and go back.

This might be a VS Code bug, and includes/defines are still resolved correctly in this case, so I guess it is only a minor annoyance. At least I can live with that, so thanks for the fix!

PS Unfortunately cppcheck issue that I raised in another discussion thread is still there in this release..
